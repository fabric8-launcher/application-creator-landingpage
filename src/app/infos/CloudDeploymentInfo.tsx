import {Grid, GridItem} from '@patternfly/react-core';
import * as React from 'react';
import {ExternalLink} from '../../shared/components/ExternalLink';

import openshiftImage from '../../assets/img/openshift.png';
import InfoCard from '../components/InfoCard';
import {CloudIcon} from '@patternfly/react-icons';

interface CloudDeploymentInfoProps {
  applicationUrl: string;
  openshiftConsoleUrl: string;
}

export function CloudDeploymentInfo(props: CloudDeploymentInfoProps) {
  return (
    <InfoCard name="cloud-deployment">
      <InfoCard.Title><CloudIcon className="with-text"/> Cloud Deployment</InfoCard.Title>
      <InfoCard.Body>
        <Grid>
          <GridItem span={3}>
            <img src={openshiftImage}/>
          </GridItem>
          <GridItem span={9}>
            <p>This system is deployed on the OpenShift Cloud Platform,
              a Kubernetes distribution which builds and runs the applications and services we've created.</p>
          </GridItem>
          <InfoCard.Separator />
          <GridItem span={3}>Application URL</GridItem>
          <GridItem span={9}>
            <ExternalLink href={props.applicationUrl}>{props.applicationUrl}</ExternalLink>
            <br/>
            The link to this application running live on OpenShift
          </GridItem>
          <InfoCard.Separator />
          <GridItem span={3}>OpenShift Console URL</GridItem>
          <GridItem span={9}>
            <ExternalLink href={props.openshiftConsoleUrl}>{props.openshiftConsoleUrl}</ExternalLink>
            <br/>
            Link to manage the OpenShift instance where this application is deployed.
          </GridItem>
        </Grid>

      </InfoCard.Body>
    </InfoCard>
  );
}