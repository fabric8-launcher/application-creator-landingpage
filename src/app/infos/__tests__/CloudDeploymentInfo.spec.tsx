import * as React from 'react';
import {shallow} from 'enzyme';
import { CloudDeploymentInfo } from '../CloudDeploymentInfo';

describe('<CloudDeploymentInfo />', () => {
  it('check that render is correct', () => {
    const component = shallow(<CloudDeploymentInfo applicationUrl="http://www.application-url.com" />);
    expect(component).toMatchSnapshot();
  });

  it('check that render is correct with repositoryUrl', () => {
    const component = shallow(<CloudDeploymentInfo applicationUrl="http://www.application-url.com" openshiftConsoleUrl="http://www.console-url.com"/>);
    expect(component).toMatchSnapshot();
  });

});

