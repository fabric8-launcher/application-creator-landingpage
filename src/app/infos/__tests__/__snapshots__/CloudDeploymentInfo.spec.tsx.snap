// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloudDeploymentInfo /> check that render is correct 1`] = `
<InfoCard
  name="cloud-deployment"
>
  <Component>
    <CloudIcon
      className="with-text"
      color="currentColor"
      size="sm"
      title={null}
    />
     Cloud Deployment
  </Component>
  <Component>
    <Grid
      className=""
      gutter={null}
      lg={null}
      md={null}
      sm={null}
      span={null}
      xl={null}
    >
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={3}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        <img />
      </GridItem>
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={9}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        <p>
          This system is deployed on the OpenShift Cloud Platform, a Kubernetes distribution which builds and runs the applications and services we've created.
        </p>
      </GridItem>
      <Component />
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={3}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        Application Name
      </GridItem>
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={9}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        <span
          className="monospace"
        />
      </GridItem>
      <Component />
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={3}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        Application URL
      </GridItem>
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={9}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        <ExternalLink
          href="http://www.application-url.com"
        >
          http://www.application-url.com
        </ExternalLink>
        <br />
        The link to this application running live on OpenShift
      </GridItem>
    </Grid>
  </Component>
</InfoCard>
`;

exports[`<CloudDeploymentInfo /> check that render is correct with repositoryUrl 1`] = `
<InfoCard
  name="cloud-deployment"
>
  <Component>
    <CloudIcon
      className="with-text"
      color="currentColor"
      size="sm"
      title={null}
    />
     Cloud Deployment
  </Component>
  <Component>
    <Grid
      className=""
      gutter={null}
      lg={null}
      md={null}
      sm={null}
      span={null}
      xl={null}
    >
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={3}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        <img />
      </GridItem>
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={9}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        <p>
          This system is deployed on the OpenShift Cloud Platform, a Kubernetes distribution which builds and runs the applications and services we've created.
        </p>
      </GridItem>
      <Component />
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={3}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        Application Name
      </GridItem>
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={9}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        <span
          className="monospace"
        />
      </GridItem>
      <Component />
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={3}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        Application URL
      </GridItem>
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={9}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        <ExternalLink
          href="http://www.application-url.com"
        >
          http://www.application-url.com
        </ExternalLink>
        <br />
        The link to this application running live on OpenShift
      </GridItem>
      <Component />
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={3}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        OpenShift Console URL
      </GridItem>
      <GridItem
        className=""
        lg={null}
        lgOffset={null}
        lgRowSpan={null}
        md={null}
        mdOffset={null}
        mdRowSpan={null}
        offset={null}
        rowSpan={null}
        sm={null}
        smOffset={null}
        smRowSpan={null}
        span={9}
        xl={null}
        xlOffset={null}
        xlRowSpan={null}
      >
        <ExternalLink
          href="http://www.console-url.com/console/project/undefined"
        >
          http://www.console-url.com/console/project/undefined
        </ExternalLink>
        <br />
        Link to manage the OpenShift instance where this application is deployed.
      </GridItem>
    </Grid>
  </Component>
</InfoCard>
`;
